src-java/orientdb-migration/src/main/java/org/openkilda/FlowMirrorMigration.java

```diff
@@ -0,0 +1,91 @@
+/* Copyright 2023 Telstra Open Source
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda;
+
+import com.orientechnologies.orient.core.db.ODatabaseSession;
+import com.orientechnologies.orient.core.db.OrientDB;
+import com.orientechnologies.orient.core.db.OrientDBConfig;
+import liquibase.change.custom.CustomTaskChange;
+import liquibase.change.custom.CustomTaskRollback;
+import liquibase.database.Database;
+import liquibase.exception.CustomChangeException;
+import liquibase.exception.RollbackImpossibleException;
+import liquibase.exception.SetupException;
+import liquibase.exception.ValidationErrors;
+import liquibase.resource.ResourceAccessor;
+
+public class FlowMirrorMigration implements CustomTaskChange, CustomTaskRollback {
+    private final String host = System.getenv("KILDA_ORIENTDB_HOST");
+    private final String dbName = System.getenv("KILDA_ORIENTDB_DB_NAME");
+    private final String user = System.getenv("KILDA_ORIENTDB_USER");
+    private final String password = System.getenv("KILDA_ORIENTDB_PASSWORD");
+    private ODatabaseSession db;
+
+    @SuppressWarnings({"UnusedDeclaration", "FieldCanBeLocal"})
+    private ResourceAccessor resourceAccessor;
+
+    @Override
+    public void execute(Database database) throws CustomChangeException {
+        String script = "BEGIN;\n";
+        script += "FOREACH ($path IN (SELECT FROM flow_mirror_path)) {\n";
+        script += "LET $fm = INSERT INTO flow_mirror (flow_mirror_id, forward_path_id, mirror_switch_id, "
+                + "egress_switch_id, status, egress_port, egress_outer_vlan, egress_inner_vlan) VALUES ($path.path_id, "
```

Comment:
This is why we need more modern java with multiline strings. 🙂
A minor thing: you can use `StringBuilder::append` instead of `+=` and `+`

Reply:
Agree, StringBuilder has better performance, but from my point of view, the current implementation is a little bit more readable and focused on the SQL script itself. If you don't mind, I would prefer to keep the current implementation.